pxtcompiler/emitter/driver.ts

```diff
@@ -297,7 +297,7 @@ namespace ts.pxtc {
             snippetMode: opts.snippetMode || false,
             alwaysEmitOnStart: opts.alwaysDecompileOnStart,
             includeGreyBlockMessages,
-            generateSourceMap: !!opts.ast,
+            generateSourceMap: opts.generateSourceMap !== undefined ? opts.generateSourceMap : !!opts.ast,
```

Comment:
i feel like this should only ever be overridden to be true, right? I don't think targets should ever be able to set this to false

Reply:
It was being set to true in my case by virtue of opts.ast being true. I wanted it turned off though, and turns out there was an existing flag for that. I wanted to preserve the original behavior when the flag was not passed in.